include/aws/cryptosdk/materials.h

```diff
@@ -472,6 +472,23 @@ struct aws_cryptosdk_keyring_vt {
         enum aws_cryptosdk_alg_id alg);
 };
 
+/**
+ * Constant time check of data-structure invariants for struct aws_cryptosdk_keyring_vt.
+ */
+AWS_CRYPTOSDK_STATIC_INLINE bool aws_cryptosdk_keyring_vt_is_valid(const struct aws_cryptosdk_keyring_vt *vtable) {
+    return AWS_OBJECT_PTR_IS_READABLE(vtable) && (vtable->vt_size == sizeof(struct aws_cryptosdk_keyring_vt)) &&
+           aws_c_string_is_valid(vtable->name);
+}
+
+/**
+ * Constant time check of data-structure invariants for struct aws_cryptosdk_keyring.
+ */
+AWS_CRYPTOSDK_STATIC_INLINE bool aws_cryptosdk_keyring_is_valid(const struct aws_cryptosdk_keyring *keyring) {
+    return AWS_OBJECT_PTR_IS_READABLE(keyring) && aws_atomic_var_is_valid(&keyring->refcount) &&
+           aws_atomic_load_int(&keyring->refcount) > 0 && aws_atomic_load_int(&keyring->refcount) <= SIZE_MAX &&
+           (keyring->vtable == NULL || aws_cryptosdk_keyring_vt_is_valid(keyring->vtable));
+}
```

Comment:
@tawdry-audrey could you confirm whether these validity functions are reasonable?

Reply:
@ttjsu-aws could you confirm whether these validity functions are reasonable?

Reply:
Yep, they seem reasonable to me.

Reply:
@ttjsu-aws We can check for nullness, but anything other than that is hard to do in plain C.